### PR TITLE
Enhance oasis-node stake account info CLI command's output

### LINKED ISSUE #3557: Show delegations in `oasis-node stake account info`
##### SUMMARY

`oasis-node stake account info` returns 0 balances on `Escrow Account` for a non-validator account.
https://www.oasisscan.com/accounts/detail/oasis1qqs6ylpfurhf6gc9mw232fkmrt3d0673lyzc5xf2
```
$ oasis-node stake account info -a $ADDR --stake.account.address oasis1qqs6ylpfurhf6gc9mw232fkmrt3d0673lyzc5xf2
General Account:
  Balance: ROSE 148689190.999985
  Nonce:   1
Escrow Account:
  Active:
    Balance:      ROSE 0.0
    Total Shares: 0
  Debonding:
    Balance:      ROSE 0.0
    Total Shares: 0
  Commission Schedule:
    Rates: (none)
    Rate Bounds: (none)
  Stake Accumulator:
    Claims: (none)
```
On the other hand, a validator account returns the correct balances.
https://www.oasisscan.com/accounts/detail/oasis1qz72lvk2jchk0fjrz7u2swpazj3t5p0edsdv7sf8
```
$ oasis-node stake account info   -a $ADDR   --stake.account.address oasis1qz72lvk2jchk0fjrz7u2swpazj3t5p0edsdv7sf8
General Account:
  Balance: ROSE 3845.994520821
  Nonce:   7
Escrow Account:
  Active:
    Balance:      ROSE 171584440.736695885
    Total Shares: 167427289728797480
  Debonding:
    Balance:      ROSE 58748.122830794
    Total Shares: 58748122830794
  Commission Schedule:
    Rates:
      (1) start: epoch 23
          rate:  10.0%
      (2) start: epoch 86
          rate:  15.0%
    Rate Bounds:
      (1) start:        epoch 0
          minimum rate: 0.0%
          maximum rate: 20.0%
  Stake Accumulator:
    Claims:
      - Name: registry.RegisterEntity
        Staking Thresholds:
          - Global: entity
      - Name: registry.RegisterNode.aJFHeID4Q7qUfMa42dRwaa9PQrZ/cVDiE3WNt4bQjo0=
        Staking Thresholds:
          - Global: node-validator
```
##### ISSUE TYPE
- Bug Report

##### OASIS NODE VERSION

```paste below
Software version: 20.12.2
Consensus:
  Consensus protocol version: 2.0.0
  Tendermint Core version:    0.34.0
  ABCI library version:       0.17.0
Runtime:
  Host protocol version:      1.0.0
  Committee protocol version: 1.0.0
Go toolchain version: 1.15.5
```

##### OS / ENVIRONMENT

`Ubuntu 20.10`

### PR DESCRIPTION
**Pull request by tjanez**
Closes #3557.

Example output for an account with nothing except some general balance:
```
$ oasis-node stake account info -a $ADDR --stake.account.address oasis1qqnhh0clvj978ym92rj07udfrku74yaluyz9e0la
Balance:
  Total: 10000.000000001 ROSE
  Available: 10000.000000001 ROSE

Nonce: 0
```

Example output for an account with a self-delegation:
```
$ oasis-node stake account info -a $ADDR --stake.account.address oasis1qqtcrdt33kxsvergtru8jm4umq8wf6tqlq3qq5zc
Balance:
  Total: 11654.352017571 ROSE
  Available: 425.337214176 ROSE
  Active Delegations from this Account:
    Total: 11229.014803395 ROSE
    Delegations:
      - To:     oasis1qqtcrdt33kxsvergtru8jm4umq8wf6tqlq3qq5zc (self)
        Amount: 11229.014803395 ROSE (10000000000000 shares)

Active Delegations to this Account:
  Total: 11229.014803395 ROSE (10000000000000 shares)
  Delegations:
    - From:   oasis1qqtcrdt33kxsvergtru8jm4umq8wf6tqlq3qq5zc (self)
      Amount: 11229.014803395 ROSE (10000000000000 shares)

Nonce: 0
```

Example output for an account with 2 outgoing delegations (one of them being a self-delegation):
```
$ oasis-node stake account info -a $ADDR --stake.account.address oasis1qzqcxqjlht6edm9mg67p4rvwtd9xqk65hqjae2zp
Balance:
  Total: 10621.867894818 ROSE
  Available: 0.0 ROSE
  Active Delegations from this Account:
    Total: 10621.867894818 ROSE
    Delegations:
      - To:     oasis1qzqcxqjlht6edm9mg67p4rvwtd9xqk65hqjae2zp (self)
        Amount: 10000.0 ROSE (10000000000000 shares)
      - To:     oasis1qr62uu00g3k95fvxyd2ydmgtyfglvfql7yxfsnld
        Amount: 621.867894818 ROSE (1000000000000 shares)

Active Delegations to this Account:
  Total: 10000.0 ROSE (10000000000000 shares)
  Delegations:
    - From:   oasis1qzqcxqjlht6edm9mg67p4rvwtd9xqk65hqjae2zp (self)
      Amount: 10000.0 ROSE (10000000000000 shares)

Nonce: 0
```

Example output for an account corresponding to an entity running a validator:
```
$ oasis-node stake account info -a $ADDR --stake.account.address oasis1qrvsa8ukfw3p6kw2vcs0fk9t59mceqq7fyttwqgx
Balance:
  Total: 10905.281231304 ROSE
  Available: 376.594833237 ROSE
  Active Delegations from this Account:
    Total: 10528.686398067 ROSE
    Delegations:
      - To:     oasis1qrvsa8ukfw3p6kw2vcs0fk9t59mceqq7fyttwqgx (self)
        Amount: 10528.686398067 ROSE (10000000000000 shares)

Active Delegations to this Account:
  Total: 10528.686398067 ROSE (10000000000000 shares)
  Delegations:
    - From:   oasis1qrvsa8ukfw3p6kw2vcs0fk9t59mceqq7fyttwqgx (self)
      Amount: 10528.686398067 ROSE (10000000000000 shares)

Stake Accumulator:
  Claims:
    - Name: registry.RegisterEntity
      Staking Thresholds:
        - Global: entity
    - Name: registry.RegisterNode.9Epy5pYPGa91IJlJ8Ivb5iby+2ii8APXdfQoMZDEIDc=
      Staking Thresholds:
        - Global: node-validator

Nonce: 0
```

Example output for an account having multiple incoming and outgoing active/debonding delegations, allowances and a commission schedule:
```
$ oasis-node stake account info -a $ADDR --stake.account.address oasis1qr62uu00g3k95fvxyd2ydmgtyfglvfql7yxfsnld
Balance:
  Total: 14769.09458142 ROSE
  Available: 50.40480345 ROSE
  Active Delegations from this Account:
    Total: 9513.013651142 ROSE
    Delegations:
      - To:     oasis1qq774msxmtghldtwn55t25fxqppdpsdyfvhnkpyt
        Amount: 4681.035394337 ROSE (4359083450983 shares)
      - To:     oasis1qr62uu00g3k95fvxyd2ydmgtyfglvfql7yxfsnld (self)
        Amount: 3460.758364288 ROSE (5565102159355 shares)
      - To:     oasis1qz8ctjdmrcgl83wh7zg5p2rsdhak6llcg52ysm8f
        Amount: 1371.219892517 ROSE (1209824320092 shares)
  Debonding Delegations from this Account:
    Total: 5205.676126828 ROSE
    Delegations:
      - To:       oasis1qr62uu00g3k95fvxyd2ydmgtyfglvfql7yxfsnld (self)
        Amount:   57.383924178 ROSE (124398043123 shares)
        End Time: epoch 1375
      - To:       oasis1qrzn0eqym8a3gkq8cc0n5ldhyyq540eumcqvnh0v
        Amount:   1070.213904708 ROSE (3890012380912 shares)
        End Time: epoch 1380
      - To:       oasis1qzmsk5v69h4c4nhuwz8pgdsyfkjjku9nxgxn0xkf
        Amount:   6.029560381 ROSE (349234823123 shares)
        End Time: epoch 1380
      - To:       oasis1qrzn0eqym8a3gkq8cc0n5ldhyyq540eumcqvnh0v
        Amount:   2626.926164403 ROSE (9548348473435 shares)
        End Time: epoch 1400
      - To:       oasis1qr62uu00g3k95fvxyd2ydmgtyfglvfql7yxfsnld (self)
        Amount:   1384.913409458 ROSE (3002243581235 shares)
        End Time: epoch 1400
      - To:       oasis1qzmsk5v69h4c4nhuwz8pgdsyfkjjku9nxgxn0xkf
        Amount:   60.2091637 ROSE (3487341581235 shares)
        End Time: epoch 1400

Allowances for this Account:
  Total: 36.85022574 ROSE
  Allowances:
    - Beneficiary: oasis1qr9gy9gx6sr9vmca5cj5484a7p4r6zssky768ngz
      Amount:      20.384938736 ROSE
    - Beneficiary: oasis1qraz5yz8r2vv57mjc4dtkgz7edj5cmhm7cktqret
      Amount:      10.482938283 ROSE
    - Beneficiary: oasis1qr62uu00g3k95fvxyd2ydmgtyfglvfql7yxfsnld (self)
      Amount:      5.982348721 ROSE

Active Delegations to this Account:
  Total: 10231.744448938 ROSE (16453244385484 shares)
  Delegations:
    - From:   oasis1qzhwl87e7d6up3tf3wu08e9n8gqk6vkw2v8gqc2j
      Amount: 6149.11818983 ROSE (9888142226129 shares)
    - From:   oasis1qr62uu00g3k95fvxyd2ydmgtyfglvfql7yxfsnld (self)
      Amount: 3460.758364288 ROSE (5565102159355 shares)
    - From:   oasis1qzqcxqjlht6edm9mg67p4rvwtd9xqk65hqjae2zp
      Amount: 621.867894818 ROSE (1000000000000 shares)

Debonding Delegations to this Account:
  Total: 5744.592227132 ROSE (12453244385484 shares)
  Delegations:
    - From:     oasis1qzhwl87e7d6up3tf3wu08e9n8gqk6vkw2v8gqc2j
      Amount:   92.258564102 ROSE (200000000000 shares)
      End Time: epoch 1365
    - From:     oasis1qqc2xclg3rxgl4kl2y0aa9nxeedwt3gjsskg4rd3
      Amount:   2414.634175931 ROSE (5234493294823 shares)
      End Time: epoch 1375
    - From:     oasis1qr62uu00g3k95fvxyd2ydmgtyfglvfql7yxfsnld (self)
      Amount:   57.383924178 ROSE (124398043123 shares)
      End Time: epoch 1375
    - From:     oasis1qr62uu00g3k95fvxyd2ydmgtyfglvfql7yxfsnld (self)
      Amount:   1384.913409458 ROSE (3002243581235 shares)
      End Time: epoch 1400
    - From:     oasis1qqc2xclg3rxgl4kl2y0aa9nxeedwt3gjsskg4rd3
      Amount:   92.258564102 ROSE (200000000000 shares)
      End Time: epoch 1400
    - From:     oasis1qzhwl87e7d6up3tf3wu08e9n8gqk6vkw2v8gqc2j
      Amount:   92.258564102 ROSE (200000000000 shares)
      End Time: epoch 1400
    - From:     oasis1qzhwl87e7d6up3tf3wu08e9n8gqk6vkw2v8gqc2j
      Amount:   1610.885025255 ROSE (3492109466303 shares)
      End Time: epoch 1450

Commission Schedule:
  Rates:
    (1) start: epoch 54
        rate:  15.0%
    (2) start: epoch 419
        rate:  19.5%
  Rate Bounds:
    (1) start:        epoch 0
        minimum rate: 0.0%
        maximum rate: 20.0%

Nonce: 3
```

### CODE REVIEW

**Review by kostko — COMMENTED** [review]
I think the API changes make things more clear and the infos query is there to avoid some round trips.

File: go/staking/tests/tester.go [hunk 1]
```diff
@@ -583,7 +583,7 @@ func testEscrowEx( // nolint: gocyclo
 	newDstAcc = nil
```
Comment (kostko) [inline go/staking/tests/tester.go:584]: There should also be some tests for all the new query methods.

Comment (tjanez) [reply go/staking/tests/tester.go:584]: Done in #3774.

File: go/common/prettyprint/quantity.go [hunk 2]
```diff
@@ -8,6 +8,91 @@ import (
...
+	}
+}
+
+// IsValid returns true iff the q is valid.
+func (q Quantity) IsValid() bool {
+	return !q.invalid && q.quan.IsValid()
+}
+
+// QuanQuantity returns q's wrapped quantity.Quantity object.
+func (q Quantity) QuanQuantity() *quantity.Quantity {
```
Comment (kostko) [inline go/common/prettyprint/quantity.go:69]: Call it `Unwrap` or `Inner`.

File: go/oasis-node/cmd/stake/delegations.go [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+				delDesc := delegationDescription{
+					delAddr,
+					delAddr.Equal(addr),
+					delegationAmount(del.Shares, sharePool),
+					del.Shares,
+					del.DebondEndTime,
+				}
+				delDescs = append(delDescs, delDesc)
+			}
+		}
```
Comment (kostko) [inline go/oasis-node/cmd/stake/delegations.go:229]: Add an explicit default case.

File: go/common/prettyprint/quantity.go (hunk 2 above)
Comment (tjanez) [reply go/common/prettyprint/quantity.go]: I've changed it to `Unwrap`.

File: go/oasis-node/cmd/stake/delegations.go (hunk 3 above)
Comment (tjanez) [reply go/oasis-node/cmd/stake/delegations.go:229]: Added.

**Review by kostko — APPROVED** [review]

**Pull request merged**